pkg/machine/fcos.go

```diff
@@ -18,32 +18,90 @@ import (
 	"github.com/coreos/stream-metadata-go/fedoracoreos"
 	"github.com/coreos/stream-metadata-go/release"
 	"github.com/coreos/stream-metadata-go/stream"
-
 	digest "github.com/opencontainers/go-digest"
 	"github.com/sirupsen/logrus"
 )
 
-// These should eventually be moved into machine/qemu as
-// they are specific to running qemu
-var (
-	artifact = "qemu"
-	Format   = "qcow2.xz"
-)
+type imageCompression int64
+type artifact int64
+type imageFormat int64
 
 const (
 	// Used for testing the latest podman in fcos
 	// special builds
 	podmanTesting     = "podman-testing"
 	PodmanTestingHost = "fedorapeople.org"
 	PodmanTestingURL  = "groups/podman/testing"
+
+	Xz imageCompression = iota
+	Zip
+	Gz
+	Bz2
+
+	Qemu artifact = iota
+	HyperV
+	None
+
+	qcow imageFormat = iota
+	vhdx
+	Tar
 )
 
+//
+// TODO artifact, imageformat, and imagecompression should be probably combined into some sort
+// of object which can "produce" the correct output we are looking for bc things like
+// image format contain both the image type AND the compression.  This work can be done before
+// or after the hyperv work.  For now, my preference is to NOT change things and just get things
+// typed strongly
+//
+
+func (a artifact) String() string {
+	if a == HyperV {
+		return "hyperv"
+	}
+	return "qemu"
+}
+
+func (imf imageFormat) String() string {
+	switch imf {
+	case vhdx:
+		return "vhdx.zip"
+	case Tar:
+		return "tar.xz"
+	}
+	return "qcow2.xz"
+}
+
+func (c imageCompression) String() string {
+	switch c {
+	case Gz:
+		return "gz"
+	case Zip:
+		return "zip"
+	case Bz2:
+		return "bz2"
+	}
+	return "xz"
+}
+
+func compressionFromFile(path string) imageCompression {
+	switch {
+	case strings.HasSuffix(path, Bz2.String()):
+		return Bz2
+	case strings.HasSuffix(path, Gz.String()):
+		return Gz
+	case strings.HasSuffix(path, Zip.String()):
+		return Zip
+	}
+	return Xz
```

Comment:
Shouldn't this actually check for "xz" suffix and error if none are found?

Reply:
meh, not really